=== modified file 'libzip/CMakeLists.txt'
--- libzip/CMakeLists.txt	2012-01-09 03:18:36 +0000
+++ libzip/CMakeLists.txt	2012-01-10 07:37:05 +0000
@@ -9,6 +9,10 @@
 
 PROJECT(libzip C)
 
+if(NOT DEFINED BUILD_SHARED_LIBS)
+    option(BUILD_SHARED_LIBS "Build a shared library form of libzip" ON)
+endif()
+
 INCLUDE(CheckFunctionExists)
 INCLUDE(CheckIncludeFiles)
 INCLUDE(CheckSymbolExists)
@@ -74,6 +78,14 @@
 ADD_DEFINITIONS("-D_CRT_SECURE_NO_WARNING")
 ENDIF(MSVC)
 
+IF(MSVC)
+  SET(CMAKE_DEBUG_POSTFIX "d")
+ENDIF(MSVC)
+
+IF(BUILD_SHARED_LIBS)
+  ADD_DEFINITIONS("-DZIP_DLL")
+ENDIF(BUILD_SHARED_LIBS)
+
 # Targets
 ADD_SUBDIRECTORY(lib)
 ADD_SUBDIRECTORY(man)

=== modified file 'libzip/lib/CMakeLists.txt'
--- libzip/lib/CMakeLists.txt	2012-01-09 03:18:36 +0000
+++ libzip/lib/CMakeLists.txt	2012-01-10 06:51:01 +0000
@@ -143,7 +143,12 @@
   SET(LIBZIP_EXTRA_FILES mkstemp.c)
 ENDIF(NOT HAVE_MKSTEMP)
 
-ADD_LIBRARY(zip SHARED ${LIBZIP_SOURCES} ${LIBZIP_EXTRA_FILES})
+IF(BUILD_SHARED_LIBS)
+	ADD_LIBRARY(zip SHARED ${LIBZIP_SOURCES} ${LIBZIP_EXTRA_FILES})
+ELSE(BUILD_SHARED_LIBS)
+	ADD_LIBRARY(zip ${LIBZIP_SOURCES} ${LIBZIP_EXTRA_FILES})
+ENDIF(BUILD_SHARED_LIBS)
+
 SET_TARGET_PROPERTIES(zip PROPERTIES VERSION 3.0 SOVERSION 3 )
 TARGET_LINK_LIBRARIES(zip ${ZLIB_LIBRARY})
 INSTALL(TARGETS zip

=== modified file 'libzip/lib/zip.h'
--- libzip/lib/zip.h	2012-01-09 03:18:36 +0000
+++ libzip/lib/zip.h	2012-01-10 07:27:35 +0000
@@ -36,12 +36,12 @@
 
 
 
+#if !defined(ZIP_EXTERN) && defined(ZIP_DLL) && defined(_WIN32)
+#	define ZIP_EXTERN __declspec(dllimport)
+#endif
+
 #ifndef ZIP_EXTERN
-#ifdef _WIN32
-#define ZIP_EXTERN __declspec(dllimport)
-#else
-#define ZIP_EXTERN
-#endif
+#	define ZIP_EXTERN extern
 #endif
 
 #ifdef __cplusplus

=== modified file 'libzip/lib/zip_utf-8.c'
--- libzip/lib/zip_utf-8.c	2012-01-09 03:18:36 +0000
+++ libzip/lib/zip_utf-8.c	2012-01-10 07:39:27 +0000
@@ -33,7 +33,6 @@
 
 
 
-#include "zip.h"
 #include "zipint.h"
 
 #include <stdlib.h>

=== modified file 'libzip/lib/zipint.h'
--- libzip/lib/zipint.h	2012-01-09 03:18:36 +0000
+++ libzip/lib/zipint.h	2012-01-10 07:39:27 +0000
@@ -36,10 +36,26 @@
 
 #include <zlib.h>
 
+/* NOTE: libzip's internal c files should always
+	include #include zipint.h, not zip.h.
+	zipint.h will include zip.h automatically,
+	and this way ZIP_EXTERN will be correctly set.
+	
+	All linking programs should #include zip.h first,
+   before zipint.h (if it is required at all).
+ */
+
 #ifdef _WIN32
-#define ZIP_EXTERN __declspec(dllexport)
-/* for dup(), close(), etc. */
-#include <io.h>
+	/* for dup(), close(), etc. */
+#	include <io.h>
+#endif
+
+#if !defined(ZIP_EXTERN) && defined(ZIP_DLL) && defined(_WIN32)
+#	define ZIP_EXTERN __declspec(dllexport)
+#endif
+
+#ifndef ZIP_EXTERN
+#	define ZIP_EXTERN extern
 #endif
 
 #include "zip.h"

